### PR TITLE
meshreg: support load gzip compressed dynamic configmap

### PR DESCRIPTION
**Pull request by believening**
The dynamically generated configurations in large-scale service data scenarios may be large enough to exceed the maximum storage capacity of configmap, making it impossible to issue dynamic configurations. 

The meshreg configuration parameters provide many runtime modification APIs for individual services and even individual instances. These APIs are mainly centered around the service name, which means that there will be a lot of repeated strings in the configuration. Compressing the configuration and saving it in configmap can increase the upper limit of the dynamic configuration size. 

Testing shows that the compression rate of the original dynamic configuration using gzip is around 10%, and supporting gzip compression can increase the upper limit of the dynamic configuration size by about 10 times.

### CODE REVIEW

File: staging/src/slime.io/slime/modules/meshregistry/module/module.go [hunk 1]
```diff
@@ -140,17 +144,44 @@ func (m *Module) prepareDynamicConfigController(opts module.ModuleOptions) (*mes
 	changeNotifyCh := make(chan struct{}, 1)
 	cmCache := &singleConfigMapCache{}
 
+	var readConfig func(cm *corev1.ConfigMap) ([]byte, bool)
+	itemKey := configMapModuleKey(opts.Env.Config.Name)
+	compressed, _ := strconv.ParseBool(os.Getenv("DYNAMIC_CONFIG_COMPRESSED"))
```
Comment (YonkaFang) [inline staging/src/slime.io/slime/modules/meshregistry/module/module.go:149]: env可以放全局变量，不用每次都取。 这里也不用全局生效，可以根据cm中文件名后缀来识别，比如`.gz`结尾。

File: staging/src/slime.io/slime/modules/meshregistry/module/module.go [hunk 2]
```diff
@@ -140,17 +144,44 @@ func (m *Module) prepareDynamicConfigController(opts module.ModuleOptions) (*mes
...
+	if compressed {
+		readConfig = func(cm *corev1.ConfigMap) ([]byte, bool) {
```
Comment (YonkaFang) [inline staging/src/slime.io/slime/modules/meshregistry/module/module.go:151]: 感觉 `itemKey` 应该作为此函数参数，而不必ref到外部var

Comment (believening) [reply staging/src/slime.io/slime/modules/meshregistry/module/module.go:151]: 之前文件 L148 在闭包函数为了计算这个 `itemKey` 同样需要 ref 外部的变量，且每次 load 都计算了 `itemKey`。考虑到这个 `itemKey` 实际上是固定的， 故在闭包为外计算了。

File: staging/src/slime.io/slime/modules/meshregistry/module/module.go (hunk 1 above)
Comment (YonkaFang) [reply staging/src/slime.io/slime/modules/meshregistry/module/module.go:149]:
```go
func isGzipBytes(data []byte) bool {
    r, err := gzip.NewReader(bytes.NewReader(data))
    if err != nil {
        return false
    }
    defer r.Close()
    return true
}
```

那就这样吧

**Pull request merged**